### ISSUE TITLE
ORA-00904: "installed_rank": invalid identifier: is Flyway NOT case insensitive?

### ISSUE THREAD

**Issue by adivardhan** [flyway/flyway#2777]
###### Which version and edition of Flyway are you using?
Flyway Community Edition 6.3.3 by Redgate

###### If this is not the latest version, can you reproduce the issue with the latest one as well?
(Many bugs are fixed in newer releases and upgrading will often resolve the issue)

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin)
Java API

###### Which database are you using (type & version)?
Oracle SQL 12.2

###### Which operating system are you using?
Linux

###### What did you do?
(Please include the content causing the issue, any relevant configuration settings, the SQL statement that failed (if relevant) and the command you ran.)

I created flyway table:
```
CREATE TABLE SCHEMA_VERSION (
    INSTALLED_RANK          INT                 NOT NULL,
    VERSION                 VARCHAR2(50),
    DESCRIPTION             VARCHAR2(200)       NOT NULL,
    TYPE                    VARCHAR2(20)        NOT NULL,
    SCRIPT                  VARCHAR2(1000)      NOT NULL,
    CHECKSUM                INT,
    INSTALLED_BY            VARCHAR2(100)       NOT NULL,
    INSTALLED_ON            TIMESTAMP           DEFAULT CURRENT_TIMESTAMP NOT NULL,
    EXECUTION_TIME          INT                 NOT NULL,
    SUCCESS                 NUMBER(1)           NOT NULL
);
```

Spring config:

```
spring:  
 flyway:
    enabled: true
    baseline-on-migrate: true
    check-location: false
    table: SCHEMA_VERSION
```

###### What did you expect to see?
I was not expecting any issue while flyway migration, but I can see an error.

###### What did you see instead?
Error:

```
Error while retrieving the list of applied migrations from Schema History table "DBX_ADMIN"."SCHEMA_VERSION"
------------------------------------------------------------------------------------------------------------
SQL State  : 42000
Error Code : 904
Message    : ORA-00904: "installed_rank": invalid identifier
```

Is this expected?
Are we not expecting case insensitivity?

PS: If I create table with double quoted column names, then it works. But why?

**Comment by MikielAgutu** [comment]
To clarify, are you manually creating the schema history table, or is that just the code that Flyway is running?

**Comment by adivardhan** [comment]
> To clarify, are you manually creating the schema history table, or is that just the code that Flyway is running?

I am **manually** creating the schema history table.

**Comment by MikielAgutu** [comment]
Why don't you let Flyway create the table for you?

At any rate, for reference, the code that creates the table for Oracle is here https://github.com/flyway/flyway/blob/master/flyway-core/src/main/java/org/flywaydb/core/internal/database/oracle/OracleDatabase.java#L102.

**Comment by adivardhan** [comment]
> Why don't you let Flyway create the table for you?
> 
> At any rate, for reference, the code that creates the table for Oracle is here https://github.com/flyway/flyway/blob/master/flyway-core/src/main/java/org/flywaydb/core/internal/database/oracle/OracleDatabase.java#L102.

Not willing flyway to create table is here https://github.com/flyway/flyway/issues/2757#issuecomment-610281874

As a client I may not have permissions to create table from code. And I find it very valid as in my organisation the ADMIN rights are not provided.

I have seen the create table code earlier and saw usage of double quotes in column names.
If I use it in oracle, then I will have to force the same DDL to infrastructure team. It could have been better if it were made case insensitive or we could skip creation from Flyway and do it manually.
But I guess in that case the queries won't work as the column names used in queries are looking for specific case  ("installed_rank" and NOT installed_rank/INSTALLED_RANK)

**Comment by juliahayward** [comment]
As an aside. this is somewhat related to #2761

**Comment by adivardhan** [comment]
> As an aside. this is somewhat related to #2761

Yes. It is closely related to mentioned issue.

So what would your suggestion be here, to make it case insensitive?

**Comment by MikielAgutu** [comment]
Related to https://github.com/flyway/flyway/issues/2350

**Comment by juliahayward** [comment]
I believe for now that the table will need to be created exactly as in the code creation script. We will revisit this soon for v7 - changing the history select/insert scripts is potentially a breaking change for other users, but at the major version we can make backwards-incompatible changes to the history table definition.

**Comment by DoodleBobBuffPants** [comment]
Closing until further activity

**Comment by DoodleBobBuffPants** [comment]
This is possibly also fixed by https://github.com/flyway/flyway/commit/c65dde15fb37775392de37ad6ef1bceb3f1c2e64

Would you be able to try and let us know?

**Comment by DoodleBobBuffPants** [comment]
This should be fixed in Flyway 7.5.4

Closing until further activity
